src/renderer/components/ConfigureServer.tsx

```diff
@@ -64,28 +65,26 @@ function ConfigureServer({
     const [showContent, setShowContent] = useState(false);
     const [waiting, setWaiting] = useState(false);
 
-    const canSave = name && url && !nameError && !urlError;
+    const [validationResult, setValidationResult] = useState<URLValidationResult>();
+    const [validating, setValidating] = useState(false);
+
+    const canSave = name && url && !nameError && !urlError && !validating;
 
     useEffect(() => {
         setTransition('inFromRight');
         setShowContent(true);
     }, []);
 
-    const checkProtocolInURL = (checkURL: string): Promise<string> => {
-        if (isValidURL(checkURL)) {
-            return Promise.resolve(checkURL);
-        }
-        return window.desktop.modals.pingDomain(checkURL).
-            then((result: string) => {
-                const newURL = `${result}://${checkURL}`;
-                setUrl(newURL);
-                return newURL;
-            }).
-            catch(() => {
-                console.error(`Could not ping url: ${checkURL}`);
-                return checkURL;
+    useEffect(() => {
+        clearTimeout(validationTimeout);
+        validationTimeout = setTimeout(() => {
+            setValidating(true);
+            window.desktop.validateServerURL(url).then((validationResult) => {
```

Comment:
`validateServerUrl` is async, so you never know how long will it take. A rare scenario would be the following (haven't seen the whole component, so may not be possible).

- User writes the url
- The first timeout (TA) ends, and this gets executed
- The user changes the url
- The new timeout (TB) ends, and this gets executed
- TB sets again validating to true
- TA finishes with validateServerURL, setting the validation result to whatever the URL was on TA
- TA sets validating to false, while TB is still running
- TB finishes with validateServerURL, and now sets the correct result to setValidationResult.

There is also the problem of trying to set the state when the component is unmounted if the component gets unmounted while `validateServerURL` is running.